### PR TITLE
Add Auto3Dseg AutoRunner

### PR DESCRIPTION
**Pull request by mingxin-zheng**
Signed-off-by: Mingxin Zheng <18563433+mingxin-zheng@users.noreply.github.com>

### Description

The all-in-one minimal runner for auto3dseg pipeline. HPO is not included because PR is still WIP.

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick --unittests  --disttests`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/apps/auto3dseg/auto_runner.py [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+                "num_images_per_batch": 2,
+                "num_epochs": 2,
+            }
+            runner.set_training_params(train_param)  # 2 epochs
+            runner.run()
+
+    Notes:
+        Expected results in the work_dir as below.
+            .
+            ├── algorithm_templates
```
Comment (Nic-Ma) [inline monai/apps/auto3dseg/auto_runner.py:53]: I think we can make this example shorter with "..." in it.

Thanks.

File: monai/apps/auto3dseg/auto_runner.py [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+logger = get_logger(module_name=__name__)
+
+
+class AutoRunner:
+    """
+    Auto3Dseg interface for minimal usage
+
+    Args:
+        data_src_cfg_name: path to a configuration file (yaml) that contains datalist, dataroot, and other params.
+                The config will be in a form of {"modality": "ct", "datalist": "path_to_json_datalist", "dataroot":
```
Comment (Nic-Ma) [inline monai/apps/auto3dseg/auto_runner.py:20]: indent is not correct.

File: monai/apps/auto3dseg/auto_runner.py [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+        algo_gen: on/off switch for algoGen
+        train: on/off switch for sequential schedule of model training
+        no_cache: if no_cache is True, it will reset the status and not use any previous results
+
+    Examples:
+
+        ..code-block:: python
+            work_dir = "./work_dir"
+            filename = "path_to_data_cfg"
+            runner = AutoRunner(data_src_cfg_filename, work_dir)
```
Comment (Nic-Ma) [inline monai/apps/auto3dseg/auto_runner.py:33]: This example is not correct, no definition for var `data_src_cfg_filename`.
Maybe `AutoRunner(data_src_cfg_filename=filename, work_dir=work_dir)`?

Thanks.

File: monai/apps/auto3dseg/auto_runner.py [hunk 4]
```diff
@@ -0,0 +1,200 @@
...
+            runner.run()
+
+        ..code-block:: python
+            work_dir = "./work_dir"
+            filename = "path_to_data_cfg"
+            runner = AutoRunner(data_src_cfg_filename, work_dir)
```
Comment (Nic-Ma) [inline monai/apps/auto3dseg/auto_runner.py:39]: Same as above.

Thanks.

File: monai/apps/auto3dseg/auto_runner.py (hunk 3 above)
Comment (mingxin-zheng) [reply monai/apps/auto3dseg/auto_runner.py:71]: Corrected. Thanks!

File: monai/apps/auto3dseg/auto_runner.py (hunk 4 above)
Comment (mingxin-zheng) [reply monai/apps/auto3dseg/auto_runner.py:77]: Corrected. Thanks!

File: monai/apps/auto3dseg/auto_runner.py (hunk 2 above)
Comment (mingxin-zheng) [reply monai/apps/auto3dseg/auto_runner.py]: Corrected. Thanks!

File: monai/apps/auto3dseg/auto_runner.py (hunk 1 above)
Comment (mingxin-zheng) [reply monai/apps/auto3dseg/auto_runner.py]: Made it shorter. Thanks!

File: monai/apps/auto3dseg/auto_runner.py (hunk 3 above)
Comment (Nic-Ma) [reply monai/apps/auto3dseg/auto_runner.py:71]: I don't see any change about this comment, have you submitted your commit?

Thanks.

**Review by Nic-Ma — APPROVED** [review]

**Pull request merged**